lib/services/local_notifications_service.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:injectable/injectable.dart';
+import 'package:logger/logger.dart';
+import 'package:mcnd_mobile/data/models/app/azan_notification_payload.dart';
+import 'package:mcnd_mobile/data/models/app/salah.dart';
+import 'package:mcnd_mobile/data/models/app/salah_time.dart';
+import 'package:timezone/timezone.dart' as tz;
+
+@lazySingleton
+class LocalNotificationsService {
+  final FlutterLocalNotificationsPlugin _plugin;
+  final Logger _logger;
+
+  LocalNotificationsService(this._plugin, this._logger);
+
+  bool _initialized = false;
+  bool get isInitialized => _initialized;
+
+  Future<void> initialize() async {
+    const androidInit = AndroidInitializationSettings('app_icon');
+
+    const iosInit = IOSInitializationSettings();
+
+    const InitializationSettings initializationSettings = InitializationSettings(
+      android: androidInit,
+      iOS: iosInit,
+    );
+
+    await _plugin.initialize(
+      initializationSettings,
+    );
+
+    await _plugin.initialize(initializationSettings);
+
+    _initialized = true;
+  }
+
+  Future<void> scheduleAzan(Salah salah, SalahTime salahTime) async {
+    final String salahName = salah.getStringName();
+
+    if (await isAzanScheduled(salah, salahTime)) {
+      _logger.i('Salah $salahName at ${salahTime.azan} is already scheduled');
+      return;
+    }
+
+    final int id = Random().nextInt(1024 * 1024);
+    final DateTime dateTime = salahTime.azan;
+    final AzanNotificationPayload payload = AzanNotificationPayload(
+      id: id,
+      salah: salah,
+      dateTime: dateTime,
+    );
+
+    await _plugin.zonedSchedule(
+      id,
+      '$salahName Azan',
+      'Time for ${salahName.toLowerCase()} salah',
+      tz.TZDateTime.from(dateTime, tz.local),
+      const NotificationDetails(
+        android: AndroidNotificationDetails(
+          'azan',
+          'Azan Notifications',
+          'MCND azan notification channel',
+        ),
+      ),
+      androidAllowWhileIdle: true,
+      uiLocalNotificationDateInterpretation: UILocalNotificationDateInterpretation.absoluteTime,
+      payload: json.encode(payload.toJson()),
+    );
+
+    _logger.i('Scheduled a notification for $salahName azan at $dateTime');
+  }
+
+  Future<bool> isAzanScheduled(Salah salah, SalahTime salahTime) async {
```

Comment:
cache results for all available notifications instead of calling the plugin every time